os/mm/mm_heap/mm_manage_allocfail.c

```diff
@@ -99,6 +105,25 @@ void mm_manage_alloc_fail(struct mm_heap_s *heap, int startidx, int endidx, size
 	abort_mode = true;
 #endif
 
+#if defined(CONFIG_DEBUG_MM_HEAPINFO)
+#if defined(CONFIG_WATCHDOG) && defined(CONFIG_MM_ASSERT_ON_FAIL)
```

Comment:
I don't know those conditionals are necessary. Let's discuss it again.

| watchdog enabled | watchdog started | reboot enabled | disable watchdog | print heapinfo |
|---|---|---|---|---|
| X | X | X | X | O |
| O | X | X | X | O |
| O | O | X | X | X |
| O | O | O | O | O |
| X | X | O | X | O |

Reply:
Thanks, I organized your table in more detail.
| watchdog enabled(config)	| reboot enabled(config) | watchdog started| watchdog stoped  | = print heapinfo|
|---|---|---|---|---|
|X|-|-|-|=O|
|O|X|X|-(must don't stop)|=O|
|O|X|O|-(must don't stop)|=X|
|O|O|X|-|=O|
|O|O|O|O|=O|

1. if watchdog is disabled, always print heapinfo. 
2. if reboot config is disabled, must not stop watchdog, but if watchdog is started, do not print heapinfo.
3. if reboot config is enabled, able to stop watchdog, print heapinfo
4. if watchdog is not started, always print heapinfo.

---
This is error case.
| watchdog enabled(config)| reboot enabled(config) | watchdog started| watchdog stoped  | = print heapinfo|
|---|---|---|---|---|
|O|X|?(error)|-(must don't stop)|=X|
|O|O|?(error)|O|=O|
|O|O|?(error)|X|=X|
|O|O|O|?(error)|=O|

1. if reboot config is disabled and we can't get watchdog state, don't print heapinfo.
2. if reboot conifg is enabled and wa can't get wtchdog state, try to stop watchdog and follow the result
3. if fail to stop watchdog, print heapinfo. below reason

* Some chipset(STM32) don't have function of stop watchdog.
try to stop watchdog, get a fail.
this case must be printed heapinfo.